read.go

```diff
@@ -440,18 +440,23 @@ func (r *reader) readNativeFrames(parsedData *Dataset, fc chan<- *frame.Frame, v
 	image.Frames = make([]*frame.Frame, nFrames)
 	bo := r.rawReader.ByteOrder()
 	pixelBuf := make([]byte, bytesAllocated)
+	nativeDataBuf := make([][]int, pixelsPerFrame)
+	buf := make([]int, pixelsPerFrame*samplesPerPixel)
 	for frameIdx := 0; frameIdx < nFrames; frameIdx++ {
+		if fc == nil {
+			nativeDataBuf = make([][]int, pixelsPerFrame)
+			buf = make([]int, pixelsPerFrame*samplesPerPixel)
+		}
 		// Init current frame
 		currentFrame := frame.Frame{
 			Encapsulated: false,
 			NativeData: frame.NativeFrame{
 				BitsPerSample: bitsAllocated,
 				Rows:          MustGetInts(rows.Value)[0],
 				Cols:          MustGetInts(cols.Value)[0],
-				Data:          make([][]int, pixelsPerFrame),
+				Data:          nativeDataBuf,
```

Comment:
question: by reusing the nativeDataBuf across multiple different frames sent to the frameChannel, doesn't this mean that data from old frames will be overwritten? We don't know for sure what the caller goroutines will do with the frames or how long they will hold onto the frames emitted from the channels, so by overwriting the underlying frame data in this goroutine, all old frames become invalid right? I'm not sure this would be a good thing to do with this API, if I'm understanding correctly.

Reply:
You're right. I'll revert it